tinygrad/tensor.py

```diff
@@ -659,7 +659,7 @@ def fix(x:Tensor): return x.reshape(*ret.shape[0:-2], ret.shape[-2] * ret.shape[
   @staticmethod
   def _tri(r:sint, c:sint, k:int=0, **kwargs) -> Tensor:
     assert all_int((r,c)), "does not support symbolic"
-    return Tensor.arange(r, **kwargs).unsqueeze(1).expand(r,c) <= Tensor.arange(-k, c-k, **kwargs).unsqueeze(0).expand(r,c)
+    return (Tensor.arange(r, **kwargs).unsqueeze(1).expand(r,c) <= Tensor.arange(-k, c-k, **kwargs).unsqueeze(0).expand(r,c)).float()
```

Comment:
this does not looks right, it's the input to where?

Reply:
when you compare two aranges they get fused and the acc becomes bool, we need sum to be at least int32
https://gist.github.com/Qazalin/38316f9c3d06be9ecb0057519333e5ed

Reply:
now it's correct
```
#include <metal_stdlib>
using namespace metal;
kernel void r_5_5(device bool* data0, const device int* data1, uint3 gid [[threadgroup_position_in_grid]], uint3 lid [[thread_position_in_threadgroup]]) {
  int gidx0 = gid.x; /* 5 */
  int acc0 = 0;
  int val0 = *(data1+0);
  for (int ridx0 = 0; ridx0 < 5; ridx0++) {
    acc0 = (((((gidx0*(-1))+(ridx0*(-1)))<(-3))?1:0)+acc0);
  }
  *(data0+gidx0) = (!((acc0+(-2))<val0));
}
```